### PR TITLE
translate use_concepts.rst and  workflow_of_capi.md

### LINKED ISSUE #8905: Translation Plan-基本使用概念-汉译英-概述
## 任务背景
针对v2版本，目前我们已有很多中文使用文档，但仍有大量的汉译英工作及少量英译汉工作没有完成，我们希望将中英文版本补充完全，提升用户的阅读体验，也有益于国外用户在使用时查阅文档。
此外，我们的模型库Models，仍有大部分英译汉和汉译英的工作，亟待完成。

## 任务描述
1. 需要做什么：
把“基本使用概念”汉译英，英文标题为Basic Concept，请勿改动
2. 地址
    - 中文地址： [Github地址](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/v2/getstarted/concepts/use_concepts_cn.rst)，[网页地址](http://www.paddlepaddle.org/docs/develop/documentation/zh/getstarted/concepts/use_concepts_cn.html)
    - 需要翻译的英文地址： [Github地址](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/v2/getstarted/concepts/use_concepts_en.rst)，[网页地址](http://www.paddlepaddle.org/docs/develop/documentation/en/getstarted/concepts/use_concepts_en.html)

## 如何认领
我们欢迎大家积极认领，内部小伙伴请通过self assign的方式认领，外部小伙伴可通过在本issue下方comment认领。

## 注意事项
上交初稿的截止日期是2018年4月15日
上交后，PaddlePaddle的RD会为大家review
请于review后的5个工作日之内完成修改

## 相关信息
所有需要翻译的文档：
V2使用文档：
https://shimo.im/sheet/PqK407VYm80Ju3dw
Models:
https://shimo.im/sheet/6XTZnRQEUS0uC1Ut

### PR DESCRIPTION
**Pull request by vienous**
fix https://github.com/PaddlePaddle/Paddle/issues/8905 and https://github.com/PaddlePaddle/Paddle/issues/8918

### CODE REVIEW

File: doc/v2/getstarted/concepts/use_concepts_en.rst [hunk 1]
```diff
@@ -134,7 +134,7 @@ Finally we can start the training by calling trainer's train method:
         event_handler=event_handler,
```
Comment (skylarch) [inline doc/v2/getstarted/concepts/use_concepts_en.rst]: PaddlePaddle是源于百度的一个深度学习平台 -> PaddlePaddle是百度的深度学习平台

中文版：这里将介绍PaddlePaddle的基本使用概念，并且展示了如何利用PaddlePaddle来解决一个经典的线性回归问题......
改为：这篇文档将介绍PaddlePaddle的基本使用概念，并展示如何利用PaddlePaddle来解决一个经典的线性回归问题

"宝塔的底端需要坚实的基座来支撑" -> 宝塔需要坚实的基座来支撑自己

”即大部分值为0，但有值的地方必须为1“ -> 即大部分元素取值为0，其他非0的元素均取值为1
"即大部分值为0，但有值的部分可以是任何浮点数。" -> 即大部分元素取值为0，其他非0的元素可以取值为任意浮点数。

File: doc/v2/getstarted/concepts/use_concepts_en.rst [hunk 2]
```diff
@@ -134,7 +134,7 @@ Finally we can start the training by calling trainer's train method:
...
         num_passes=100)
 
-For more information on using the PaddlePaddle, please refer to `Advanced guide <../../howto/index_cn.html>`_。
+For more information on using the PaddlePaddle, please refer to `advanced guide <../../howto/index_cn.html>`_。
 
 Linear regression complete example
 ==============
```
Comment (skylarch) [inline doc/v2/getstarted/concepts/use_concepts_en.rst]: ############
Basics
 ############

PaddlePaddle is a deep learning platform developed by Baidu. PaddlePaddle provides a rich API for deep learning researchers. It easily perform tasks such as neural network configuration and model training.
Here, we will introduce the basic concepts of PaddlePaddle and show how to use PaddlePaddle to solve a classic linear regression problem.
Before using this document, please refer to `Installation Documentation <../../build_and_install/index_cn.html>`_ to complete the installation of PaddlePaddle.


Network Configuration
============

Loading PaddlePaddle
----------------------

Before configuring the network, you first need to load the corresponding Python library and perform initialization.

.. code-block:: bash

    Import paddle.v2 as paddle
    Import numpy as np
    Paddle.init(use_gpu=False)


Building a Neural Network
-----------------------

Building a neural network is like using a building block to build a pagoda. In PaddlePaddle, the layer is our building block, and the neural network is the pagoda we want to build. We combine different layers to build a neural network.
The pagoda needs a solid base to support itself. Similarly, the neural network needs some specific layers as the input interfaces to complete the network training.

For example, we can define a layer to describe the neural network input as follows:

.. code-block:: bash

    x = paddle.layer.data(name='x', type=paddle.data_type.dense_vector(2))
    y = paddle.layer.data(name='y', type=paddle.data_type.dense_vector(1))

Variable `x` indicates that the input data is a two-dimensional dense vector, and `y` indicates that the input data is a one-dimensional dense vector.

PaddlePaddle supports four input data types and three input sequence modes.

Four data types:

Dense_vector: Dense floating-point vector.
* sparse_binary_vector: a vector with most elements 0 and all other non-zero elements 1;
* sparse_float_vector: a vector with most elements 0 and all other non-zero elements any floating-point number;
* integer

Three sequence modes:

* SequenceType.NO_SEQUENCE: not a sequence
* SequenceType.SEQUENCE: a time series
* SequenceType.SUB_SEQUENCE: a time series with each element also a time series.

Different data types and sequence modes have different return formats. The formats are as follows:

+----------------------+---------------------+---- -------------------------------+------------------ ------------------------------+
| | NO_SEQUENCE | SEQUENCE | SUB_SEQUENCE |
+======================+=====================+==== ===============================+================== ==============================+
| dense_vector | [f, f, ...] | [[f, ...], [f, ...], ...] | [[[[f, ...], ...], [[f, ...], ...],...] |
+----------------------+---------------------+---- -------------------------------+------------------ ------------------------------+
| sparse_binary_vector | [i, i, ...] | [[i, ...], [i, ...], ...] | [[[i, ...], ...], [[i, ...], ...],...] |
+----------------------+---------------------+---- -------------------------------+------------------ ------------------------------+
| sparse_float_vector | [(i,f), (i,f), ...] | [[(i,f), ...], [(i,f), ...], ...] [[[(i,f), ...], ...], [[(i,f), ...], ...],...] |
+----------------------+---------------------+---- -------------------------------+------------------ ------------------------------+
| integer_value | i | [i, i, ...] | [[i, ...], [i, ...], ...] |
+----------------------+---------------------+---- -------------------------------+------------------ ------------------------------+

In the chart, `f` represents a floating-point number and `i` represents an integer.

Note: For sparse_binary_vector and sparse_float_vector, PaddlePaddle stores the index with the value position. E.g,

- For a 5-dimensional non-sequential sparse 01 vector ``[0, 1, 1, 0, 0]``, whose type is sparse_binary_vector, which returns ``[1, 2]``.
- For a 5-dimensional non-sequential sparse floating-point vector ``[0, 0.5, 0.7, 0, 0]``, whose type is sparse_float_vector, which returns ``[(1, 0.5), (2, 0.7)] ``.


After defining the input layer, we can use other layers to combine. When combining, you need to specify the input source of the layer.

For example, we can define the following layer combination:

.. code-block:: bash

    Y_predict = paddle.layer.fc(input=x, size=1, act=paddle.activation.Linear())
    Cost = paddle.layer.square_error_cost(input=y_predict, label=y)

Among them, x and y are the input layer described before; y_predict is to receive x as an input, followed by a fully connected layer; cost receives y_predict and y as input, and is connected to a squared error layer.

The last layer of cost records all the topological structures of the neural network. By combining different layers, we can complete the construction of the neural network.


Training model
============

After completing the construction of the neural network, we first need to create parameters that need to be optimized according to the neural network structure, and create an optimizer.
After that, we can create a trainer to train the network.

.. code-block:: bash

    Parameters = paddle.parameters.create(cost)
    Optimizer = paddle.optimizer.Momentum(momentum=0)
    Trainer = paddle.trainer.SGD(cost=cost,
                                 Parameters=parameters,
                                 Update_equation=optimizer)

Among them, the trainer receives three parameters, including neural network topology, neural network parameters, and iterative equations.

In the process of building a neural network, we only describe the input to the neural network. The trainer needs to read the training data. The PaddlePaddle uses the reader to load the data.

.. code-block:: bash

    #define training dataset reader
    Def train_reader():
        Train_x = np.array([[1, 1], [1, 2], [3, 4], [5, 2]])
        Train_y = np.array([[-2], [-3], [-7], [-7]])
        Def reader():
            For i in xrange(train_y.shape[0]):
                Yield train_x[i], train_y[i]
        Return reader

Finally, we can start the training by calling the :code:`train` method in :code:`trainer`:

.. code-block:: bash

    # define feeding map
    Feeding = {'x': 0, 'y': 1}

    # event_handler to print training info
    Def event_handler(event):
        If isinstance(event, paddle.event.EndIteration):
            If event.batch_id % 1 == 0:
                Print "Pass %d, Batch %d, Cost %f" % (
                    Event.pass_id, event.batch_id, event.cost)
    # training
    Trainer.train(
        Reader=paddle.batch(train_reader(), batch_size=1),
        Feeding=feeding,
        Event_handler=event_handler,
        Num_passes=100)

For more information on how to use the PaddlePaddle, please refer to the `Advanced Steps <../../howto/index_cn.html>`_.

Solving a Linear Regression Problem with PaddlePaddle
==============================================

Here is an example of using linear regression to fit a straight line in three-dimensional space:

.. literalinclude:: src/train.py
    :linenos:

Use the above trained model to predict. Take one of the models :code:`params_pass_90.tar`, input the vector group that needs to be predicted, and print the output:

.. literalinclude:: src/train.py
     :linenos:

For the practical application of linear regression, refer to the first chapter of the PaddlePaddle book <http://book.paddlepaddle.org/index.html>`_.

**Review by skylarch — CHANGES_REQUESTED** [review]
